### PR TITLE
Allow import/export of grok patterns in content packs

### PR DESCRIPTION
**Pull request by joschi**
Previously it wasn't possible to export or import grok patterns as part of a content pack. This could lead to broken extractors in inputs.

This PR adds the capabilities to import/export grok patterns in content packs.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/grok/GrokPattern.java [hunk 1]
```diff
@@ -17,28 +17,29 @@
...
-        return "GrokPattern{" +
-                "id=" + id +
-                ", name='" + name + '\'' +
-                ", pattern='" + pattern + '\'' +
-                '}';
+        return MoreObjects.toStringHelper(this)
+                .add("id", id)
+                .add("name='", name)
+                .add("pattern='", pattern)
+                .add("contentPack", contentPack)
```
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/grok/GrokPattern.java]: Tiny thing, but we should remove those `='` from name and pattern, as `MoreObjects.toStringHelper` will take care of that

**Comment by edmundoa** [comment]
There are a couple of issues I found:

- Importing a content pack with grok patterns ignores them. I think the changes are missing the `grokPatterns` list in `org/graylog2/restclient/models/api/requests/CreateBundleRequest.java`.

- We don't establish dependencies between extractors using grok patterns and grok patterns, so it is possible at the moment to export an extractor that uses a grok pattern, but not that pattern. Should we resolve those dependencies?

**Comment by joschi** [comment]
> We don't establish dependencies between extractors using grok patterns and grok patterns, so it is possible at the moment to export an extractor that uses a grok pattern, but not that pattern. Should we resolve those dependencies?

I don't think that this is necessary now. In order to properly handle those dependencies, we'd need to introduce a (very) special handling for the GrokExtractor and analyze all used patterns recursively (as grok patterns can include other grok patterns) and then export all those required patterns. If people start complaining, we can still implement it, but for now I wouldn't want to put in those extra hours for this PR.

**Comment by edmundoa** [comment]
Another alternative would be to export all grok patterns, but it's also not an optimal solution. Anyway, I think this is good enough and much better than what we had before!

LGTM after the last changes :shipit:

**Pull request merged**